### ISSUE TITLE
v0.1.8 does not support dustjs v2.7

### ISSUE THREAD

**Issue by longtian** [krakenjs/adaro#61]
Problem is 
https://github.com/krakenjs/kraken-js/issues/371

Btw, unit test fails in this combination
- adaro@v0.1.8 
- dustjs v2.7

But, this will pass
- adaro@v0.1.8 
- dustjs v2.6.2

**Comment by sethkinast** [comment]
`dust.load` isn't part of the exposed API anymore. I'ma take a look at how we can refactor the code that needs to patch it (dustjacket).

**Comment by aredridel** [comment]
Oh dear. Alright! I really wish dust would adopt semver.

**Comment by pvenkatakrishnan** [comment]
+1 to semver.

**Comment by sethkinast** [comment]
For better or for worse, Dust versioning policy is such that you should lock to a minor version (not a major) to avoid breakages.

e.g. `"dustjs-linkedin": "~2.6"`

Maybe it can change in Dust 3...

**Comment by sethkinast** [comment]
Although dustjacket (and adaro) don't explicitly lock Dust versions anywhere so I am not sure you would be saved anyways :P

**Comment by aredridel** [comment]
Indeed. We demand apps do that (and that's what npm3 will pay attention to) -- peer dependencies are becoming advisory.

**Comment by ricardohbin** [comment]
The peerDependencies of dustjs-helpers can cause troubles too. Make sure to downgrade it too ->

```
    "dustjs-helpers": "~1.6"
    "dustjs-linkedin": "~2.6.0"
```

**Comment by aredridel** [comment]
Yep! Well aware of that one. Dust versioning is a mess.

**Comment by ricardohbin** [comment]
This commit https://github.com/linkedin/dustjs/commit/885d7e21af3d284c6f97a67901d08c8ee8f5956e#diff-6837aa089aead29d68bfb62936171c2bL93 :P

**Comment by aredridel** [comment]
Yup! https://github.com/linkedin/dustjs/pull/641 is a good attempt at adding the infrastructure to do an equivalent to what we need.

**Comment by aredridel** [comment]
Fixing this will be a breaking change, so I'm preparing an adaro 1.0 release.

**Comment by aredridel** [comment]
Fixed in #69
